plugins/out_s3/s3.c

```diff
@@ -981,6 +981,7 @@ static int upload_data(struct flb_s3 *ctx, struct s3_file *chunk,
                        char *body, size_t body_size,
                        const char *tag, int tag_len)
 {
+    struct flb_input_instance *i_ins;
```

Comment:
This won't be set to anything... it won't be initialized... you want the `i_ins` from the `cb_s3_flush` arguments. You need to find some way to pass that down to this code, this may require some refactoring of funciton signatures to add the new argument.